src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.service;
+
+import core.basesyntax.service.operation.OperationHandler;
+import core.basesyntax.service.operation.OperationStrategy;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private Map<String, Integer> storage = new HashMap<>();
```

Comment:
The `storage` field should not be in the `ShopServiceImpl` class. It breaks the principle of Single Responsibility. It would be better to create a separate class `StorageService` and move the `storage` field there. This is a code quality issue.